pkg/controller/cluster/cluster_controller.go

```diff
@@ -130,6 +130,10 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (_ reconcile.Res
 		// Remove when https://github.com/kubernetes-sigs/controller-runtime/issues/526 is fixed.
 		cluster.SetGroupVersionKind(gvk)
 		if err := r.Client.Status().Patch(ctx, cluster, patchCluster); err != nil {
+			if apierrors.IsNotFound(err) {
```

Comment:
Can you add a comment here that this is expected during deletion, since the finalizer is removed during the previous patch command?

Reply:
@detiber updated, thanks.